packages/@tinacms/toolkit/src/packages/fields/plugins/BlocksFieldPlugin.tsx

```diff
@@ -94,9 +95,21 @@ const Blocks = ({ tinaForm, form, field, input }: BlockFieldProps) => {
     [field.name, form.mutators]
   )
 
+  const showFilter = React.useMemo(() => {
+    return Object.entries(field.templates).length > 6
+  }, [field.templates])
+  const [filter, setFilter] = React.useState('')
+  const filteredBlocks = React.useMemo(() => {
+    return Object.entries(field.templates).filter(([name, template]) => {
+      return (
+        template.label.toLowerCase().includes(filter.toLowerCase()) ||
```

Comment:
If `label` is null (which I think it can be?) I think this would cause an error

Reply:
Just added a ternary here & on the label output to fallback on name alone.